lua/neogit/lib/collection.lua

```diff
@@ -0,0 +1,42 @@
+local M = {}
+
+function M.keyBy(tbl, key)
```

Comment:
I'm not really a fan of the name as it was a bit confusing at the beginning without looking at the code. Could you change this to a more specific name (and also try to keep the names snake case like the rest of the code base)?

Reply:
Sure on the snake case (guess I got confused for a second there). How would you call it though? I only know the method by that name ([here](https://laravel.com/docs/8.x/collections#method-keyby) is where I got it from), so I'm kinda biased here.

Reply:
Hmm, maybe this is more of a convention thing? I guess if you just add some comments explaining what this does it would be good 👍